### DISCUSSION TITLE
Sync doesn't complete entire directory structure in one go

### DISCUSSION THREAD

**Discussion by cm0rR** [abraunegg/onedrive#1591] [Q&A] [answered]
Using `onedrive --synchronize --upload-only` I find that sometimes it will sync a larger amount of files in one go (300+), and sometimes a smaller amount (68).  However it won't sync the entire directory tree and contents - is there a reason for this?

Folder structure is ~13,000 files in ~900 folders.
Local folder source is an CIFS mount.
No error is displayed, the last file complete's and then no more are tried.  When re-running the command immediately after, another small batch sync's.
```
onedrive version                       = v2.4.13-7-g2b32c3a
Config path                            = /home/chris/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /onedrive
Config option 'skip_dir'               =
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = true
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false

[chris@al log]$ uname -r
5.13.5-arch1-1

[chris@al ~]$ swapon --show
NAME      TYPE      SIZE USED PRIO
/dev/vda2 partition 1.8G   2M   -2

[chris@al ~]$ df -h
Filesystem            Size  Used Avail Use% Mounted on
dev                   983M     0  983M   0% /dev
run                   991M  620K  990M   1% /run
/dev/vda1              38G   11G   25G  31% /
tmpfs                 991M     0  991M   0% /dev/shm
tmpfs                 991M  8.0K  991M   1% /tmp
tmpfs                 199M   52K  198M   1% /run/user/1000
//192.168.0.210/data  237G  123G  114G  52% /onedrive/data
```

**Comment by abraunegg** [comment, marked as answer]
@cm0rR 
This 99% sounds like this: https://github.com/abraunegg/onedrive/blob/master/docs/known-issues.md#application-stops-running-without-any-visible-reason

When ever something like this is popping up here are the common elements:
1. Users are usually running Ubuntu of some sort, or a Ubuntu variant
2. Users are based in Europe, accessing the Eastern European Microsoft Data Centre

Most likely:
* Something on / within your Linux OS is the most likely cause
* Your Internet Router .. plausible, but doubtful this is the cause
* Your ISP .. plausible, but doubtful
* Your network path between you and Microsoft was somehow disrupted .. plausible, but then the reports would be everywhere of an issue 
* A change at Microsoft .. plausible, but then significantly more people would be impacted

Things for you to check:
* Exit code of the application when you are seeing this
* Perform the HTTPS debugging to determine why the exit is occurring
* Analyse your network between you and OneDrive
* Some folk, when using `onedrive --monitor` have found setting the interval to 200 rather than 300 helps - this indicates that something on the local Linux OS is having an issue with TCP wait states of 5 mins .. but this is not your issue given you are using `onedrive --synchronize`

It must also be noted that I am *never ever* able to reproduce this, nor create / replicate an environment where this issue occurs.

If this was a bug in the application - this would be reported hundreds of times a day .. the only viable items are (as per above):
* Something on / within your Linux OS is the most likely cause
* Your network path between you and Microsoft was somehow disrupted

**Reply by cm0rR** [reply]
@abraunegg after your (swift!) response, I moved this function to a CentOS server and am finding much better sync results which do not artificially terminate - so it seems your "something to do with OS" comments were the likely cause

Appreciate the feedback, and keep up the good work!
